### ISSUE TITLE
🐞 CFA Descriptor.forward interpolates raw backbone features instead of pooled features

### ISSUE THREAD

**Issue by wsora** [open-edge-platform/anomalib#3454]
### Summary
The CFA `Descriptor.forward()` implementation appears to regress compared to **v0.7.0**: when fusing multi-scale backbone features, the code interpolates and concatenates **raw** feature maps instead of the **pooled** feature maps. This changes the descriptor feature space and can significantly degrade training / evaluation metrics even when hyperparameters and data config are unchanged.

### Environment
- **anomalib version**: [e.g. 1.0.1]
- **PyTorch version**: [e.g. 2.x]
- **torchvision version**: [e.g. 0.x]
- **Python version**: [e.g. 3.10]
- **OS**: [e.g. Ubuntu 22.04]

### Expected behavior
When combining features from multiple backbone layers, each branch should use the **same preprocessing** as in v0.7.0: apply `avg_pool2d` first, then `interpolate` the **pooled** tensor to match the reference spatial size before concatenation.

### Actual behavior
The current code interpolates the **original** `feature` tensor (pre-pooling) for non-first branches, which is inconsistent with v0.7.0 and likely incorrect for the intended multi-scale fusion.

### Minimal reproduction / pointer to code
File: `src/anomalib/models/image/cfa/torch_model.py`  
Class: `Descriptor.forward(...)`

**v0.7.0 behavior (reference)**  
Loop variable after pooling is used for interpolation/concatenation (pooled representation).

**Current behavior (suspected bug)**  
`F.interpolate(feature, patch_features.size(2), ...)` uses the raw feature map.

**Suggested fix**  
Use `pooled_features` for interpolation, e.g. `F.interpolate(pooled_features, patch_features.size(2), mode="bilinear")`.

### Impact
- Training loss may still optimize, but **metrics can drop noticeably** because the descriptor / memory bank statistics change.
- Reproducibility: users comparing **0.7.0 vs 2.x** with identical configs may observe worse CFA results.

### Dataset

MVTecAD

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

```
from anomalib.data import MVTecAD
from anomalib.models import Cfa
from anomalib.engine import Engine

# Initialize components
datamodule = MVTecAD(root="/mnt/share/wgd/other/DataSets/MVTecAD/mvtec_anomaly_detection",
                     category="grid")

model = Cfa()

engine = Engine(max_epochs=30)

# Train the model
engine.fit(datamodule=datamodule, model=model)
# Test the model
engine.test(model=model, datamodule=datamodule)
```

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.6274018287658691     │
│       image_F1Score       │    0.8484848737716675     │
│        pixel_AUROC        │    0.9640755653381348     │
│       pixel_F1Score       │    0.3038134276866913     │
└───────────────────────────┴───────────────────────────┘
With the above code, I can only get the following result. (image_AUROC 0.62 instead of 0.998)

### OS information

OS information:
- OS: Ubuntu 20.04
- Python version: 3.10.0
- Anomalib version: 2.3.0dev0
- PyTorch version: pytorch-lightning 2.6.1
- CUDA/cuDNN version: 12.8
- GPU models and configuration: 2x GeForce RTX 3090Ti

### Expected behavior

When combining features from multiple backbone layers, each branch should use the **same preprocessing** as in v0.7.0: apply `avg_pool2d` first, then `interpolate` the **pooled** tensor to match the reference spatial size before concatenation.

### Pip/GitHub

pip

### What version/branch did you use?

2.3.0dev0

### Configuration YAML

```yaml
model:
  class_path: anomalib.models.Cfa
  init_args:
    backbone: wide_resnet50_2
    gamma_c: 1
    gamma_d: 1
    num_nearest_neighbors: 3
    num_hard_negative_features: 3
    radius: 1.0e-05

trainer:
  max_epochs: 30
  callbacks:
    - class_path: lightning.pytorch.callbacks.EarlyStopping
      init_args:
        patience: 5
        monitor: pixel_AUROC
        mode: max
```

### Logs

```shell
/home/wgd/anaconda3/envs/anomalib3/bin/python /mnt/share/wgd/other/anomalib_yamlConfig/anomalib/xxx.py 
/home/wgd/anaconda3/envs/anomalib3/lib/python3.10/site-packages/timm/models/layers/__init__.py:49: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
/home/wgd/anaconda3/envs/anomalib3/lib/python3.10/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/wgd/anaconda3/envs/anomalib3/lib/python3.10/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=Wide_ResNet50_2_Weights.IMAGENET1K_V1`. You can also use `weights=Wide_ResNet50_2_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
 Tip: For seamless cloud logging and experiment tracking, try installing [litlogger](https://pypi.org/project/litlogger/) to enable LitLogger, which logs metrics and artifacts automatically to the Lightning Experiments platform.
You are using a CUDA device ('NVIDIA GeForce RTX 3090 Ti') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [1]
┏━━━┳━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━┳━━━━━━━┓
┃   ┃ Name           ┃ Type          ┃ Params ┃ Mode  ┃ FLOPs ┃
┡━━━╇━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━╇━━━━━━━┩
│ 0 │ pre_processor  │ PreProcessor  │      0 │ train │     0 │
│ 1 │ post_processor │ PostProcessor │      0 │ train │     0 │
│ 2 │ evaluator      │ Evaluator     │      0 │ train │     0 │
│ 3 │ model          │ CfaModel      │ 31.3 M │ train │     0 │
│ 4 │ loss           │ CfaLoss       │      0 │ train │     0 │
└───┴────────────────┴───────────────┴────────┴───────┴───────┘
Trainable params: 31.3 M                                                        
Non-trainable params: 0                                                         
Total params: 31.3 M                                                            
Total estimated model params size (MB): 125                                     
Modules in train mode: 22                                                       
Modules in eval mode: 121                                                       
Total FLOPs: 0                                                                  
/home/wgd/anaconda3/envs/anomalib3/lib/python3.10/site-packages/lightning/pytorch/loops/fit_loop.py:534: Found 121 module(s) in eval mode at the start of training. This may lead to unexpected behavior during training. If this is intentional, you can ignore this warning.
100%|██████████| 9/9 [00:03<00:00,  2.56it/s]
`Trainer.fit` stopped: `max_epochs=30` reached.
Epoch 29/29 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9/9 0:00:04 • 0:00:00 3.13it/s 
The following callbacks returned in `LightningModule.configure_callbacks` will override existing callbacks passed to Trainer: Evaluator, ImageVisualizer, PostProcessor, PreProcessor
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [1]
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.6115288138389587     │
│       image_F1Score       │          0.84375          │
│        pixel_AUROC        │    0.9632418155670166     │
│       pixel_F1Score       │    0.30601567029953003    │
└───────────────────────────┴───────────────────────────┘
Testing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3/3 0:00:03 • 0:00:00 0.65it/s 

Process finished with exit code 0

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
